src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java

```diff
@@ -246,7 +246,8 @@ private static void verifyHasProductionFile(Map<String, List<String>> allTests,
                 }
 
                 if (!path.contains(File.separatorChar + "grammar" + File.separatorChar)
-                        && !path.contains(File.separatorChar + "internal" + File.separatorChar)) {
+                        && !path.contains(File.separatorChar + "internal" + File.separatorChar)
+                        && !path.contains(File.separatorChar + "filters" + File.separatorChar)) {
```

Comment:
Please revert this.
We can put internal test input to existing folder "internal".